process/prefixer.go

```diff
@@ -32,19 +34,40 @@ func (p *Prefixer) Write(data []byte) (n int, err error) {
 	offset := 0
 	out := make([]byte, 0, len(data))
 
-	// loop through newlines
-	for offset < len(data) {
-		next := bytes.IndexRune(data[offset:], '\n')
+	// loop through line breaks and add prefix
+	for l := len(data); offset < l; {
+		// find either a newline or an escape char
+		next := bytes.IndexAny(data[offset:], "\n\x1b")
```

Comment:
It would be nice to standardise on hex `\x1b` or octal `\033` for escape chars in the codebase/docs/etc.
Currently there’s a bit of both in our codebase. Octal is a ridiculous radix. So this `\x1b` is good. Basically ignore this comment.